ddsrecorder/test/blackbox/mcap/McapFileCreationTest.cpp

```diff
@@ -0,0 +1,712 @@
+            send_data.get()
+            )()
+        );
+    pubsubType.serialize(send_data.get(), &payload);
+
+    mcap::McapReader mcap_reader;
+    auto messages = get_msgs_mcap(file_name, mcap_reader);
+
+    for (auto it = messages.begin(); it != messages.end(); it++)
+    {
+        auto received_msg = reinterpret_cast<unsigned char const*>(it->message.data);
+        for (unsigned int i = 0; i < payload.length; i++)
+        {
+            ASSERT_EQ(payload.data[i], received_msg[i]) << "wrong data !!";
+        }
+        ASSERT_EQ(payload.length, it->message.dataSize) << "length fails !!";
+    }
+    mcap_reader.close();
+
+}
+
+TEST(McapFileCreationTest, mcap_data_topic)
+{
+
+    std::string file_name = "output_mcap_data_topic_.mcap";
+
+    record(file_name);
+
+    mcap::McapReader mcap_reader;
+    auto messages = get_msgs_mcap(file_name, mcap_reader);
+
+    std::string received_topic;
+    std::string received_data_type_name;
+
+    for (auto it = messages.begin(); it != messages.end(); it++)
+    {
+        received_topic = it->channel->topic;
+        received_data_type_name = it->schema->name;
+    }
+    mcap_reader.close();
+
+    // Test data
+    ASSERT_EQ(received_topic, test::topic);
+    ASSERT_EQ(received_data_type_name, test::data_type_name);
+
+}
+
+TEST(McapFileCreationTest, mcap_data_num_msgs)
+{
+
+    std::string file_name = "output_mcap_data_num_msgs_.mcap";
+
+    record(file_name, test::n_msgs);
+
+    mcap::McapReader mcap_reader;
+    auto messages = get_msgs_mcap(file_name, mcap_reader);
+
+    unsigned int n_received_msgs = 0;
+    for (auto it = messages.begin(); it != messages.end(); it++)
+    {
+        n_received_msgs++;
+    }
+    mcap_reader.close();
+
+    // Test data
+    ASSERT_EQ(test::n_msgs, n_received_msgs);
+
+}
+
+TEST(McapFileCreationTest, mcap_data_num_msgs_downsampling)
+{
+
+    std::string file_name = "output_mcap_data_num_msgs_downsampling_.mcap";
+
+    record(file_name, test::n_msgs, test::downsampling);
+
+    mcap::McapReader mcap_reader;
+    auto messages = get_msgs_mcap(file_name, mcap_reader);
+
+    unsigned int n_received_msgs = 0;
+    for (auto it = messages.begin(); it != messages.end(); it++)
+    {
+        n_received_msgs++;
+    }
+    mcap_reader.close();
+
+    // Test data
+    unsigned int expected_msgs = test::n_msgs / test::downsampling;
+    if (test::n_msgs % test::downsampling)
+    {
+        expected_msgs++;
+    }
+    ASSERT_EQ(expected_msgs, n_received_msgs);
+
+}
+
+//////////////////////
+// With transitions //
+//////////////////////
+
+TEST(McapFileCreationTest, transition_paused_running)
+{
+    std::string file_name = "output_transition_paused_running_.mcap";
+
+    unsigned int n_data_1 = rand() % 10 + 1;
+    unsigned int n_data_2 = rand() % 10 + 1;
+
+    auto recording = record_with_transitions(
+        file_name,
+        McapHandlerState::PAUSED,
+        n_data_1, n_data_2,
+        McapHandlerState::RUNNING,
+        0);
```

Comment:
Please use false instead, this is rather confusing for the reader. And maybe set false as default value for this parameter.